google/cloud/internal/async_retry_loop_test.cc

```diff
@@ -55,46 +58,66 @@ std::unique_ptr<BackoffPolicy> TestBackoffPolicy() {
       .clone();
 }
 
+using TimerResult = StatusOr<std::chrono::system_clock::time_point>;
+
 class AsyncRetryLoopCancelTest : public ::testing::Test {
  protected:
-  int CancelCount() {
-    std::lock_guard<std::mutex> lk(mu_);
-    int n = cancel_count_;
-    cancel_count_ = 0;
-    return n;
-  }
+  std::size_t CancelCount() { return sequencer_.CancelCount(); }
```

Comment:
FYI: With the changes from #7784, this will need to become something like:
```cc
std::size_t RequestCancelCount() { return sequencer_.CancelCount("Request"); }
...
TEST(blah, blah) {
  EXPECT_EQ(0, RequestCancelCount());
}
```
or
```cc
std::size_t CancelCount(std::string name) { return sequencer_.CancelCount(name); }
....
TEST(blah, blah) {
  EXPECT_EQ(0, CancelCount("Request"));
}
```

and you could also check the timer's cancel count.